great_expectations/experimental/datasources/filesystem_data_asset.py

```diff
@@ -0,0 +1,223 @@
+from __future__ import annotations
+
+import copy
+import logging
+import pathlib
+from typing import TYPE_CHECKING, List, Optional, Pattern, Set, Tuple
+
+import pydantic
+from typing_extensions import ClassVar
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.core.batch_spec import PathBatchSpec
+from great_expectations.experimental.datasources.interfaces import (
+    Batch,
+    BatchRequest,
+    BatchRequestOptions,
+    DataAsset,
+    TestConnectionError,
+)
+
+if TYPE_CHECKING:
+    from great_expectations.execution_engine import ExecutionEngine
+
+logger = logging.getLogger(__name__)
+
+
+class _FilesystemDataAsset(DataAsset):
+    # Pandas specific class attrs
+    _EXCLUDE_FROM_READER_OPTIONS: ClassVar[Set[str]] = {
+        "name",
+        "base_directory",
+        "regex",
+        "order_by",
+    }
+
+    # Filesystem specific attributes
+    base_directory: pathlib.Path
+    regex: Pattern
+
+    # Internal attributes
+    _unnamed_regex_param_prefix: str = pydantic.PrivateAttr(
+        default="batch_request_param_"
+    )
+
+    class Config:
+        """
+        Need to allow extra fields for the base type because pydantic will first create
+        an instance of `_FilesystemDataAsset` before we select and create the more specific
+        asset subtype.
+        Each specific subtype should `forbid` extra fields.
+        """
+
+        extra = pydantic.Extra.allow
+
+    def get_reader_method(self) -> str:
+        raise NotImplementedError(
+            """One needs to explicitly provide 'reader_method' for File-Path style DataAsset extensions as temporary \
+work-around, until "type" naming convention and method for obtaining 'reader_method' from it are established."""
+        )
+
+    def get_reader_options_include(self) -> set[str] | None:
+        raise NotImplementedError
+
+    def test_connection(self) -> None:
+        """Test the connection for the CSVAsset.
+
+        Raises:
+            TestConnectionError
```

Comment:
Can you update this to say: `TestConnectionError: If the connection test fails.`. I am doing that in my open PR as well.

Reply:
@NathanFarmer Done -- in the next commit -- thank you!.